state/cleanup.go

```diff
@@ -1677,7 +1678,7 @@ func (st *State) obliterateUnit(ctx context.Context, store objectstore.ObjectSto
 		}
 	}
 	if err == nil && removed {
-		err = applicationService.DeleteUnit(ctx, unitName)
+		err = applicationService.DeleteUnit(ctx, coreunit.Name(unitName))
```

Comment:
**todo:** If it is safe to not une NewName here (and validate), add a comment that state it.